addons/account_tax_python/models/account_tax.py

```diff
@@ -54,6 +55,23 @@ def _ascending_process_taxes_batch(self, batch):
         if not batch['price_include'] and batch['amount_type'] == 'code':
             batch['computed'] = True
 
+    @api.model
+    def _eval_taxes_computation_prepare_product_fields(self, tax_values_list):
+        # EXTENDS 'account'
+        field_names = super()._eval_taxes_computation_prepare_product_fields(tax_values_list)
+        Product = self.env['product.product']
+        for tax_values in tax_values_list:
+            if tax_values['amount_type'] == 'code':
+                tax = self.browse(tax_values['id'])
+                for formula in ((tax.python_applicable or '').strip(), (tax.python_compute or '').strip()):
+                    groups = re.findall(r'((?:product\[\')(?P<field>\w+)(?:\'\]))+', formula) or []
```

Comment:
Now that we're in a double loop, what about compiling the regex before?
Did you check if we really need the `or []` at the end?